src/spicy/runtime-support.h

```diff
@@ -929,36 +920,35 @@ inline ValPtr to_val(const T& t, TypePtr target) {
     return rval;
 }
 
-/**
- * Converts a Spicy-side enum to a Zeek record value. The result is returned
- * with ref count +1.
- */
-template<typename T, typename std::enable_if_t<std::is_enum<typename T::Value>::value>*>
-inline ValPtr to_val(const T& t, TypePtr target) {
-#if SPICY_VERSION_NUMBER >= 10700
-    auto proto = typename T::Value(t.value());
-#else
-    auto proto = t;
-#endif
-
-    return to_val(proto, target);
+/** Maps HILTI's `Protocol` enum to Zeek's `transport_proto` enum. */
+inline ValPtr to_val_for_transport_proto(int64_t val, const TypePtr& target) {
+    switch ( val ) {
+        case hilti::rt::Protocol::TCP: return id::transport_proto->GetEnumVal(::TransportProto::TRANSPORT_TCP);
+        case hilti::rt::Protocol::UDP: return id::transport_proto->GetEnumVal(::TransportProto::TRANSPORT_UDP);
+        case hilti::rt::Protocol::ICMP: return id::transport_proto->GetEnumVal(::TransportProto::TRANSPORT_ICMP);
+        default: return id::transport_proto->GetEnumVal(::TransportProto::TRANSPORT_UNKNOWN);
```

Comment:
It's unfortunate that nothing makes explicit that we map `Undef` to `TRANSPORT_UNKNOWN`. If we keep accepting an `int64_t` here I'd at least add a fallthrough so we have it in the source (and can e.g., see it if we search for references),

```suggestion
        case hilti::rt::Protocol::Undef: [[fallthrough]];
        default: return id::transport_proto->GetEnumVal(::TransportProto::TRANSPORT_UNKNOWN);
```

If we'd instead accepted a `Protocol::Value` we could get rid of the `default` case altogether.

Reply:
> If we'd instead accepted a Protocol::Value we could get rid of the default case altogether.

Actually our enum types are over int so we need to keep the `default`.